Sources/BrowserServicesKit/FeatureFlagger/FeatureFlagger.swift

```diff
@@ -173,6 +277,60 @@ public class DefaultFeatureFlagger: FeatureFlagger {
         }
     }
 
+    public func getAllActiveExperiments() -> Experiments {
+        var activeExperiments = [String: ExperimentData]()
+        guard let enrolledExperiments = experimentManager?.experiments else { return activeExperiments }
+        let config = privacyConfigManager.privacyConfig
+
+        for (subfeatureID, experimentData) in enrolledExperiments {
+            let state = config.stateFor(subfeatureID: subfeatureID, parentFeatureID: experimentData.parentID)
+            guard state == .enabled || state == .disabled(.targetDoesNotMatch) else { continue }
+            let cohorts = config.cohorts(subfeatureID: subfeatureID, parentFeatureID: experimentData.parentID) ?? []
+            let experimentSubfeature = ExperimentSubfeature(parentID: experimentData.parentID, subfeatureID: subfeatureID, cohorts: cohorts)
+
+            if experimentManager?.resolveCohort(for: experimentSubfeature, isAssignCohortEnabled: false) == experimentData.cohortID {
+                activeExperiments[subfeatureID] = experimentData
+            }
+        }
+        return activeExperiments
+    }
+
+    public func getCohortIfEnabled<Flag: FeatureFlagExperimentDescribing>(for featureFlag: Flag) -> (any FlagCohort)? {
+        switch featureFlag.source {
+        case .disabled:
+            return nil
+        case .internalOnly(let cohort):
+            return cohort
+        case .remoteDevelopment where !internalUserDecider.isInternalUser:
+            return nil
+        case .remoteReleasable(let featureType),
+                .remoteDevelopment(let featureType) where internalUserDecider.isInternalUser:
+            if case .subfeature(let subfeature) = featureType {
+                if let resolvedCohortID = getCohortIfEnabled(subfeature) {
+                    return Flag.CohortType.allCases.first { return $0.rawValue == resolvedCohortID }
+                }
+            }
+            return nil
+        default:
+            return nil
+        }
+    }
+
+    private func getCohortIfEnabled(_ subfeature: any PrivacySubfeature) -> CohortID? {
```

Comment:
We could make this public too in case experiment developers don’t want to create the ExperimentFeatureFlag struct and could call this directly.
Although I think we probably should encourage the other use… it’s nicer and gives access to default internal user to a cohort and in the future overrides...

Reply:
personally I wouldn't if we want to introduce better patterns